metapackages/auto-instrumentations-node/src/utils.ts

```diff
@@ -100,52 +37,49 @@ const InstrumentationMap: Record<InstrumentationMapKeys, any> = {
   '@opentelemetry/instrumentation-koa': KoaInstrumentation,
   '@opentelemetry/instrumentation-ioredis': IORedisInstrumentation,
   '@opentelemetry/instrumentation-mongodb': MongoDBInstrumentation,
-  // '@opentelemetry/instrumentation-mysql': MySQLInstrumentation,
   '@opentelemetry/instrumentation-pg': PgInstrumentation,
-  // '@opentelemetry/instrumentation-redis': RedisInstrumentation,
+  // '@opentelemetry/instrumentation-mysql': MySQLInstrumentation,
+  // '@opentelemetry/instrumentation-redis':RedisInstrumentation,
 };
 
-const defaultConfigs: InstrumentationConfigMap = {
-  '@opentelemetry/instrumentation-dns': { enabled: true },
-  '@opentelemetry/instrumentation-express': { enabled: true },
-  '@opentelemetry/instrumentation-http': { enabled: true },
-  '@opentelemetry/instrumentation-graphql': { enabled: true },
-  '@opentelemetry/instrumentation-grpc': { enabled: true },
-  '@opentelemetry/instrumentation-koa': { enabled: true },
-  '@opentelemetry/instrumentation-ioredis': { enabled: true },
-  '@opentelemetry/instrumentation-mongodb': { enabled: true },
-  // '@opentelemetry/instrumentation-mysql': { enabled: true },
-  '@opentelemetry/instrumentation-pg': { enabled: true },
-  // '@opentelemetry/instrumentation-redis': { enabled: true }
+// Config types inferred automatically from the first argument of the constructor
+type ConfigArg<T> = T extends new (...args: infer U) => any ? U[0] : never;
+export type InstrumentationConfigMap = {
+  [Name in keyof typeof InstrumentationMap]?: ConfigArg<
+    typeof InstrumentationMap[Name]
+  >;
 };
 
 export function getNodeAutoInstrumentations(
   inputConfigs: InstrumentationConfigMap = {}
 ): Instrumentation[] {
-  const configs: InstrumentationConfigMap = Object.assign(
-    {},
-    defaultConfigs,
-    inputConfigs
-  );
-  const keys = Object.keys(configs);
+  for (const name of Object.keys(inputConfigs)) {
+    if (!Object.prototype.hasOwnProperty.call(InstrumentationMap, name)) {
+      diag.error(`Provided instrumentation name "${name}" not found`);
+      continue;
+    }
+  }
+
   const instrumentations: Instrumentation[] = [];
-  keys.forEach(key => {
-    const Instance = InstrumentationMap[key as keyof InstrumentationConfigMap];
-    const config = configs[key as keyof InstrumentationConfigMap];
-    if (Instance) {
-      try {
-        if (config?.enabled === false) {
-          diag.debug(`Disabling instrumentation for ${key}`);
-        } else {
-          diag.debug(`Loading instrumentation for ${key}`);
-          instrumentations.push(new Instance(config));
-        }
-      } catch (e) {
-        diag.error(e);
-      }
-    } else {
-      diag.error(`Provided instrumentation name "${key}" not found`);
+
+  for (const name of Object.keys(InstrumentationMap) as Array<
+    keyof typeof InstrumentationMap
+  >) {
+    const Instance = InstrumentationMap[name];
+    // Defaults are defined by the instrumentation itself
+    const userConfig = inputConfigs[name] ?? {};
```

Comment:
@obecny I left the default config out of this plugin entirely. If `enabled` is undefined the plugin is enabled so it shouldn't be a problem. All plugin configs should be optional